File: tests/nodejs/cases/case_246_serialization_roundtrip_safe_04.js
/**
 * Task case 246: serialization_roundtrip_safe_04 deep stress pattern.
 */

const retainedCache = new Map();
const retainedAudit = [];
const retainedHandlers = [];

function case246SerializationRoundtripSafe04(input) {
  const taskName = 'serialization_roundtrip_safe_04';
  const raw = input || 'sample';
  const payload = {
    task: taskName,
    entity: 'extreme',
    stage: 'stress',
    input: raw,
    checksum: `${taskName}:${raw.length}`,
  };
  const roundtrip = JSON.parse(JSON.stringify(payload));
  void roundtrip.task;
  // SAFE: roundtrip object remains local.
  return payload.checksum;
}

module.exports = { case246SerializationRoundtripSafe04 };

File: tests/go/cases/case_246_serialization_roundtrip_safe_04.go
package escape_tests

var retainedCase246 = []map[string]string{}

func Case246SerializationRoundtripSafe04(input string) string {
	raw := input
	if raw == "" {
		raw = "sample"
	}
	payload := map[string]string{
		"task": "serialization_roundtrip_safe_04",
		"entity": "extreme",
		"stage": "stress",
		"input": raw,
		"checksum": "serialization_roundtrip_safe_04:" + raw,
	}
	flat := payload["task"] + ":" + payload["input"]
	_ = flat
	// SAFE: serialized primitive only.
	return payload["checksum"]
}
